CMakeLists.txt

```diff
@@ -71,6 +88,54 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 # Default to -O2 on release builds.
 string(REPLACE "-O3" "-O2" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
 
+# Disable log for release-type builds.
+if(NOT CMAKE_C_FLAGS_RELEASE MATCHES DDISABLE_LOG)
+  set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -DDISABLE_LOG")
+endif()
+if(NOT CMAKE_C_FLAGS_MINSIZEREL MATCHES DDISABLE_LOG)
+  set(CMAKE_C_FLAGS_MINSIZEREL "${CMAKE_C_FLAGS_MINSIZEREL} -DDISABLE_LOG")
+endif()
+if(NOT CMAKE_C_FLAGS_RELWITHDEBINFO MATCHES DDISABLE_LOG)
+  set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -DDISABLE_LOG")
+endif()
+
+# Set build flags for custom Dev build type.
+# -DNDEBUG purposely omitted because we want assertions.
+if(MSVC)
+  SET(CMAKE_C_FLAGS_DEV ""
+      CACHE STRING "Flags used by the compiler during development (optimized, but with debug info and logging) builds."
+      FORCE)
+  SET(CMAKE_EXE_LINKER_FLAGS_DEV ""
+      CACHE STRING "Flags used for linking binaries during development (optimized, but with debug info and logging) builds."
+      FORCE)
+  SET(CMAKE_SHARED_LINKER_FLAGS_DEV ""
+      CACHE STRING "Flags used by the shared libraries linker during development (optimized, but with debug info and logging) builds."
+      FORCE)
+else()
+  if(CMAKE_COMPILER_IS_GNUCC)
+    check_c_compiler_flag(-Og HAS_OG_FLAG)
+  else()
+    set(HAS_OG_FLAG 0)
+  endif()
+
+  if(HAS_OG_FLAG)
+    set(CMAKE_C_FLAGS_DEV "-Og -g -DDISABLE_LOG"
+        CACHE STRING "Flags used by the compiler during development (optimized, but with debug info and logging) builds."
+        FORCE)
+  else()
+    set(CMAKE_C_FLAGS_DEV "-O2 -g -DDISABLE_LOG"
+        CACHE STRING "Flags used by the compiler during development (optimized, but with debug info and logging) builds."
+        FORCE)
+  endif()
+
+  SET(CMAKE_EXE_LINKER_FLAGS_DEV ""
+      CACHE STRING "Flags used for linking binaries during development (optimized, but with debug info and logging) builds."
+      FORCE)
+  SET(CMAKE_SHARED_LINKER_FLAGS_DEV ""
+      CACHE STRING "Flags used by the shared libraries linker during development (optimized, but with debug info and logging) builds."
```

Comment:
Can this code that is duplicated inside the `if(MSVC)` block be done outside the condition to avoid the duplicate code?